crates/turborepo-lib/src/engine/mod.rs

```diff
@@ -201,7 +197,19 @@ impl Engine<Built> {
                     }
                 }
 
-                Ok(is_persistent)
+                // check if the package for the task has that task in its package.json
+                let info = package_graph
+                    .workspace_info(&WorkspaceName::Other(task_id.package().to_string()))
+                    .expect("package graph should contain workspace info for task package");
+
+                let package_has_task = info.package_json.scripts.contains_key(task_id.task());
```

Comment:
In order to match what we do other places to match Go behavior I think we need to also treat `""` as a package not having a task.
```suggestion
                let package_has_task = info.package_json.scripts.get(task_id.task()).map_or(false, |script| !script.is_empty());
```